crates/standalone/src/subcommands/extract_schema.rs

```diff
@@ -0,0 +1,71 @@
+use std::path::PathBuf;
+
+use anyhow::Context;
+use clap::{ArgMatches, CommandFactory, FromArgMatches};
+use spacetimedb::host::extract_schema;
+use spacetimedb::messages::control_db;
+use spacetimedb_lib::{sats, RawModuleDef};
+
+/// Extracts the module schema from a local module file.
+#[derive(clap::Parser)]
+#[command(name = "extract-schema")]
+struct Args {
+    /// The module file
+    module: PathBuf,
+
+    /// The type of module
+    #[arg(long)]
+    host_type: Option<HostType>,
+}
+
+#[derive(clap::ValueEnum, Copy, Clone)]
+enum HostType {
+    Wasm,
+}
+
+impl From<HostType> for control_db::HostType {
+    fn from(x: HostType) -> Self {
+        match x {
+            HostType::Wasm => control_db::HostType::Wasm,
+        }
+    }
+}
+
+impl HostType {
+    fn from_extension(ext: &str) -> Option<Self> {
+        match ext {
+            "wasm" => Some(Self::Wasm),
+            _ => None,
+        }
+    }
+}
+
+pub fn cli() -> clap::Command {
+    Args::command()
+}
+
+pub async fn exec(args: &ArgMatches) -> anyhow::Result<()> {
```

Comment:
We should make this command as unstable.

Reply:
I was thinking that since we don't really support using `spacetimedb-standalone` by itself anyway, it probably doesn't matter much.